lisa/tools/hibernation_setup.py

```diff
@@ -126,18 +125,43 @@ def _install(self) -> bool:
         make.make_install(code_path)
         return self._check_exists()
 
-    def _check(self, pattern: Pattern[str]) -> int:
-        cat = self.node.tools[Cat]
-        log_output = ""
+    def _check(self, pattern: str) -> int:
+        """
+        Check for pattern matches in log files using grep for efficiency.
+        This avoids reading large log files entirely which can cause timeouts.
+        """
+        grep = self.node.tools[Grep]
         ls = self.node.tools[Ls]
+
+        # Determine which log file to use
         if ls.path_exists("/var/log/syslog", sudo=True):
-            log_output = cat.read("/var/log/syslog", force_run=True, sudo=True)
+            log_file = "/var/log/syslog"
         elif ls.path_exists("/var/log/messages", sudo=True):
-            log_output = cat.read("/var/log/messages", force_run=True, sudo=True)
+            log_file = "/var/log/messages"
         else:
+            # Fall back to journalctl for systems without traditional log files
             journalctl = self.node.tools[Journalctl]
             log_output = journalctl.first_n_logs_from_boot(no_of_lines=0)
-        matched_lines = find_patterns_in_lines(log_output, [pattern])
-        if not matched_lines:
+            # Compile pattern only when needed for journalctl path
+            compiled_pattern = re.compile(pattern)
+            matched_lines = find_patterns_in_lines(log_output, [compiled_pattern])
+            if not matched_lines:
+                return 0
```

Comment:
You can implement grep tool to support text.

Reply:
You mean pass the text to the function instead of file, perform a match and return result?